src/include/extension/extension_manager.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "loaded_extension.h"
+#include "main/client_context.h"
+#include "main/db_config.h"
+#include "storage/storage_extension.h"
+
+namespace kuzu {
+namespace extension {
+
+class ExtensionManager {
+public:
+    void loadExtension(const std::string& path, main::ClientContext* context);
+
+    KUZU_API std::string toCypher();
+
+    KUZU_API void addExtensionOption(std::string name, common::LogicalTypeID type,
+        common::Value defaultValue);
+
+    const main::ExtensionOption* getExtensionOption(std::string name) const;
+
+    KUZU_API void registerStorageExtension(std::string name,
+        std::unique_ptr<storage::StorageExtension> storageExtension);
+
+    common::case_insensitive_map_t<std::unique_ptr<storage::StorageExtension>>&
```

Comment:
By looking at the usage of this function, I think you don't need to output `unique_ptr` here (I don't see the caller needs to actually own the `StorageExtension`, instead would it be safer to output a reference or raw pointer of `StorageExtension`?